### PR TITLE
MAINT: Update environment.yml to match *_requirements.txt

### PR DESCRIPTION
**Pull request by melissawm**
No real reason other than I noticed the docs were being built locally on python 3.9 instead of 3.11, which is used in CI. If any of those are incorrect please let me know.

### CODE REVIEW

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks Melissa. This doesn't look quite right to me. The single `=` is the correct comparison symbol, `==` is not correct. Semantics are not the same compared to PyPI/pip specifiers (e.g., this will install 3.11.0 rather than 3.11.6). The added exact pins are not desired I think.

The only change that seems okay to make here is Python 3.11.x instead of 3.9.x

**Comment by melissawm** [comment]
Of course you're right, sorry about that !

File: environment.yml [hunk 1]
```diff
@@ -7,12 +7,12 @@ name: numpy-dev
...
   - conda-forge
 dependencies:
-  - python=3.9 #need to pin to avoid issues with builds
+  - python=3.11 #need to pin to avoid issues with builds
   - cython>=3.0
   - compilers
   - openblas
   - nomkl
-  - setuptools=59.2.0
+  - setuptools
```
[the remark this answers is not in the document]
Comment (rgommers) [reply environment.yml:15]: We still have some compile tests using `setup.py` (e.g. `_core/tests/test_limited_api.py`), so it makes sense. `setuptools` is installing with `python` by conda anyway, so in practice it doesn't matter much.

**Review by rgommers — APPROVED** [review]
LGTM now, thanks Melissa.

**Pull request merged**